### PR TITLE
audit: new formulae should not have options

### PR DESCRIPTION
**Pull request by commitay**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

### CODE REVIEW

**Comment by commitay** [comment]
Unrelated to this PR: do we want have audit check for option deps in new formulae?

**Comment by MikeMcQuaid** [comment]
This makes sense to me but I'd want @ilovezfs to 👍. Thanks @commitay!

> Unrelated to this PR: do we want to have audit check for optional deps in new formulae?

I think it kinda is related 😁 as `:optional` and `:recommended` end up being options.

Also, ideally this check would only apply to Homebrew/homebrew-core formulae.

**Comment by commitay** [comment]
> ideally this check would only apply to Homebrew/homebrew-core formulae

Updated.

File: Library/Homebrew/dev-cmd/audit.rb [hunk 1]
```diff
@@ -381,6 +381,11 @@ def audit_deps
           if dep.tags.include?(:run)
             problem "Dependency '#{dep.name}' is marked as :run. Remove :run; it is a no-op."
           end
+
+          next unless @new_formula && @official_tap
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/audit.rb]: Can you make this an `if &&`; `unless &&` is quite hard to parse.

File: Library/Homebrew/dev-cmd/audit.rb [hunk 2]
```diff
@@ -382,7 +382,9 @@ def audit_deps
             problem "Dependency '#{dep.name}' is marked as :run. Remove :run; it is a no-op."
           end
 
-          next unless @new_formula && @official_tap
+          next unless @new_formula
+          next if formula.versioned_formula?
```
Comment (commitay) [inline Library/Homebrew/dev-cmd/audit.rb]: Added this to match the `options_cop` check so it doesn't fail for deps and not options.

File: Library/Homebrew/dev-cmd/audit.rb [hunk 3]
```diff
@@ -532,7 +539,7 @@ def audit_specs
       end
 
       if @new_formula && formula.head
-        problem "New formulae should not have a HEAD spec"
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/audit.rb]: @ilovezfs Is this what you're wanting? I wasn't sure.

**Comment by MikeMcQuaid** [comment]
Just FYI @commitay https://github.com/Homebrew/brew/pull/3835 is the approach we're moving towards for new formulae-only audits so they will post a comment on the PR but not fail CI. I could do with some help testing that and getting it merged if you're interested. We can chat more about it on Slack if so (as it'll involved internal project security stuff).

File: Library/Homebrew/dev-cmd/audit.rb (hunk 3 above)
Comment (ilovezfs) [reply Library/Homebrew/dev-cmd/audit.rb]: Yes. The original wording was spelling out exactly how to cheat the audit. Open a new formula, get it merged. Then immediately open a second PR to add head.

**Comment by commitay** [comment]
> I could do with some help testing that and getting it merged if you're interested.

Definitely interested. 👍

**Comment by commitay** [comment]
@MikeMcQuaid Do you want this merged before https://github.com/Homebrew/brew/pull/3835?

**Comment by MikeMcQuaid** [comment]
I'd probably rather #3835 before this.

**Comment by commitay** [comment]
Rebased on master and updated to use `new_formula_problem`
